cloudml-template/examples/babyweight-custom/trainer/model.py

```diff
@@ -147,65 +129,61 @@ def _model_fn(features, labels, mode):
                                                 activation_fn=tf.nn.relu,
                                                 weights_initializer=tf.contrib.layers.xavier_initializer())
 
-        # Create output (logits) layer given the hidden layers
+        # Create output (logits) layer given the hidden layers (usually without applying any activation functions)
         logits = tf.layers.dense(inputs=hidden_layers,
                                  units=output_layer_size,
                                  activation=None)
 
-        # Reshape output layer to 1-dim Tensor to return predictions
-        output = tf.squeeze(logits)
-
-        # Specify the model output (i.e. predictions) given the output layer
-        predictions = {
-            'scores': output
-        }
-
-        # Specify the export output based on the predictions
-        export_outputs = {
-            'predictions': tf.estimator.export.PredictOutput(predictions)
-        }
-
-        loss = None
-        train_op = None
-        eval_metric_ops = None
-
-        if mode == tf.estimator.ModeKeys.TRAIN or mode == tf.estimator.ModeKeys.EVAL:
-            # Calculate loss using mean squared error
-            loss = tf.losses.mean_squared_error(labels, output)
-
-        if mode == tf.estimator.ModeKeys.TRAIN:
-            # Create Optimiser
-            optimizer = tf.train.AdamOptimizer(
-                learning_rate=task.HYPER_PARAMS.learning_rate)
-
-            # Create training operation
-            train_op = optimizer.minimize(
-                loss=loss, global_step=tf.train.get_global_step())
-
-        if mode == tf.estimator.ModeKeys.EVAL:
-            # Specify root mean squared error as additional eval metric
-            eval_metric_ops = {
-                "rmse": tf.metrics.root_mean_squared_error(labels, output)
-            }
-
-        # Provide an estimator spec
-        estimator_spec = tf.estimator.EstimatorSpec(mode=mode,
-                                                    loss=loss,
-                                                    train_op=train_op,
-                                                    eval_metric_ops=eval_metric_ops,
-                                                    predictions=predictions,
-                                                    export_outputs=export_outputs
-                                                    )
-        return estimator_spec
+        return logits
+
+    def _train_op_fn(loss):
```

Comment:
we typically define train_op_fn inside the model_fn. Reason: many times it needs to access tensors produces within the model_fn.